src/components/ActivityCard/ActivityCards.js

```diff
@@ -61,6 +83,10 @@ export default function ActivityCards({activities, onSetActivities, onSetIsHidde
             type={type}
             infos={infos}
             onToggleFavorites={onToggleFavorites}
+            onDelete={() => deleteActivity(id)}
```

Comment:
You do add the `id` here. In your function you need to accept the `id` as a parameter:
```js
function deleteActivity(id) {
  ....
}
```
Here I would just pass an `onDelete` function and then add the `deleteActivity` function to the `App.js`:
```suggestion
            onDelete={() => {
              onDeleteActivity(id)
              setShowModalConfirmation({
                 show: false,
                 id: null,
               });
            }}
```